Neos.Neos/Classes/FrontendRouting/DimensionResolution/Resolver/UriPathResolver.php

```diff
@@ -168,26 +175,45 @@ public function fromRequestToDimensionSpacePoint(
         $firstUriPathSegment = array_shift($uriPathSegments);
 
         if (isset($this->uriPathToDimensionSpacePoint[$firstUriPathSegment])) {
+            if (count($uriPathSegments) === 0 && $this->uriPathToDimensionSpacePoint[$firstUriPathSegment]->hash === $this->defaultDimensionSpacePoint->hash) {
+                #/en sould not match
+                return $context;
+            }
             // match
+            //die("MATCH");
             $context = $context->withRemainingUriPath('/' . implode('/', $uriPathSegments));
             $context = $context->withAddedDimensionSpacePoint(
                 $this->uriPathToDimensionSpacePoint[$firstUriPathSegment]
             );
         } elseif (isset($this->uriPathToDimensionSpacePoint[''])) {
             // Fall-through empty match (if configured)
             $context = $context->withAddedDimensionSpacePoint($this->uriPathToDimensionSpacePoint['']);
+        } else {
+            // TODO FIX ME: Context does not match -> so the default kicks in. -> Not what we want, because prefix was configured.
+            // TODO: clashes with composability!?!?
+
+            if ($firstUriPathSegment === "") {
+                # /: fill with default dsp if not found
+                $context = $context->withAddedDimensionSpacePoint($this->defaultDimensionSpacePoint);
+            }
         }
 
         return $context;
     }
 
     public function fromDimensionSpacePointToUriConstraints(
         DimensionSpacePoint $dimensionSpacePoint,
-        SiteNodeName $targetSiteIdentifier,
+        DocumentNodeInfo $targetNode,
         UriConstraints $uriConstraints
     ): UriConstraints {
         $dimensionSpacePoint = $this->reduceDimensionSpacePointToConfiguredDimensions($dimensionSpacePoint);
 
+        if ($targetNode->getUriPath() === '' && $dimensionSpacePoint->hash === $this->defaultDimensionSpacePoint->hash) {
```

Comment:
we need the target node, because UriConstraints does NOT contain the target URL...